AntWars/GamePanel.cs

```diff
@@ -60,23 +60,29 @@ partial class GamePanel : Form {
 
         private void setColor(BoardObject obj, Bitmap bitmap) {
             if (obj.isCarry()) {
-                setColor(bitmap, obj, Color.DarkGreen, Color.Blue, (obj as Ant).Owner);
+                setColor(bitmap, obj.Coords, Color.DarkGreen, Color.Blue, (obj as Ant).Owner);
             } else if (obj.isScout()) {
-                setColor(bitmap, obj, Color.Green, Color.DarkBlue, (obj as Ant).Owner);
+                setColor(bitmap, obj.Coords, Color.Green, Color.DarkBlue, (obj as Ant).Owner);
             } else if (obj.isWarrior()) {
-                setColor(bitmap, obj, Color.Red, Color.DarkViolet, (obj as Ant).Owner);
+                setColor(bitmap, obj.Coords, Color.Red, Color.DarkViolet, (obj as Ant).Owner);
             } else if (obj.isBase()) {
-                setColor(bitmap, obj, Color.GreenYellow, Color.BlueViolet, (obj as Base).Player);
+                Base playerbase = obj as Base;
+                if (playerbase.RangeLevel > 0) {
+                    foreach (Coordinates coords in playerbase.Coords.getAdjacentCoordinates(playerbase.RangeLevel)) {
```

Comment:
``playerbase.Coords.getAdjacentCoordinates(playerbase.RangeLevel)`` wird jedesmal neu-berechnet, obwohl sich die Koordinate der Base nicht ändert, das sollte zwischengespeichert werden.
Es wäre außerdem gut wenn du nur einen Array benutzt mit einem for-loop, dass das weniger Performance intensiv. Die Berechnung kannst du mit der Liste machen und dann ``#ToArray()`` sowie in der ``CoordsInView``/``CircleCalculator``.